src/lib/coda_lib/coda_lib.ml

```diff
@@ -429,6 +420,36 @@ let create (config : Config.t) =
                 in
                 Root_prover.prove ~logger:config.logger ~frontier
                   consensus_state )
+              ~get_transition_chain_witness:(fun query ->
```

Comment:
I would advise returning a reason why the request was not answered instead of just an option (for future logging/debugging). This applies to both `get_transition_chain_witness` and `get_transition_chain`.

Reply:
Don't see this logging